websites/I/IGDB/presence.ts

```diff
@@ -0,0 +1,91 @@
+      presenceData.details = "Adding a new game";
+    else {
+      delete presenceData.largeImageKey;
+      presenceData.details = `Viewing: ${document.title}`;
+      presenceData.state = document.querySelector<HTMLHeadingElement>(
+        "#content-page > div.loaded > div > div.gamepage-header-info > div.gamepage-summary > div.gamepage-title-container > div > a > h3"
+      ).textContent;
+      presenceData.largeImageKey = document.querySelector<HTMLImageElement>(
+        "#content-page > div.loaded > div > div.gamepage-header-info > div.gamepage-cover > img"
+      ).src;
+      presenceData.smallImageKey = "logo";
+      presenceData.smallImageText = "IGDB";
+    }
+  } else if (path.startsWith("/genres")) {
+    path === "/genres"
+      ? (presenceData.details = "Browsing all genres")
+      : ((presenceData.details = "Browsing a genre"),
+        (presenceData.state = document.title.slice(0, -5)));
+  } else if (path.startsWith("/platforms")) {
+    path === "/platforms"
+      ? (presenceData.details = "Browsing all platforms")
+      : (delete presenceData.largeImageKey,
+        (presenceData.details = "Viewing a platform"),
+        (presenceData.state = document.title),
+        (presenceData.largeImageKey = document.querySelector<HTMLImageElement>(
+          "#content-page > div > div > div.row > div.col-sm-4 > img"
+        ).src),
+        (presenceData.smallImageKey = "logo"),
+        (presenceData.smallImageText = "IGDB"));
+  } else if (path.startsWith("/top-100")) {
+    presenceData.details = "Viewing Top 100";
+    presenceData.state = document.title;
+  } else if (path.startsWith("/reviews"))
+    presenceData.details = "Finding game reviews";
+  else if (path.startsWith("/themes")) {
+    path === "/themes"
+      ? (presenceData.details = "Browsing all themes")
+      : ((presenceData.details = "Browsing a theme"),
+        (presenceData.state = document.title.slice(0, -5)));
+  } else if (path.startsWith("/collections")) {
+    path === "/collections"
+      ? (presenceData.details = "Browsing all collections")
+      : ((presenceData.details = "Browsing a collection"),
+        (presenceData.state = document.title));
+  } else if (path.startsWith("/player_perspectives")) {
+    path === "/player_perspectives"
+      ? (presenceData.details = "Browsing all player perspectives")
+      : ((presenceData.details = "Browsing a player perspective's games"),
+        (presenceData.state = document.title.slice(0, -5)));
+  } else if (path.startsWith("/franchises")) {
+    path === "/franchises"
+      ? (presenceData.details = "Browsing all franchises")
+      : ((presenceData.details = "Browsing a franchise"),
+        (presenceData.state = document.title));
+  } else if (path.startsWith("/categories")) {
+    path === "/categories"
+      ? (presenceData.details = "Browsing all categories")
+      : ((presenceData.details = "Browsing a category"),
+        (presenceData.state = document.title.slice(0, -5)));
```

Comment:
plz no, dont do it like this. just make it in if case and not all these (), (), ();

Reply:
It can't be that bad right lol, I'll rewrite the code when I can

Reply:
done